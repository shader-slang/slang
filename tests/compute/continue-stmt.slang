//TEST(compute):COMPARE_COMPUTE: -shaderobj
//TEST(compute):COMPARE_COMPUTE:-cpu -shaderobj

// Test that `break` from a loop works.

int test(int inVal)
{
	int ii = 0;
	do
	{
		if(ii < inVal)
		{
		    ii++;
			continue;
		}
		break;
	}
	while(true);

	return -ii;
}

//TEST_INPUT:ubuffer(data=[0 1 2 3], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer : register(u0);

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
	uint tid = dispatchThreadID.x;
	int inVal = outputBuffer[tid];
	int outVal = test(inVal);
	outputBuffer[tid] = outVal;
}