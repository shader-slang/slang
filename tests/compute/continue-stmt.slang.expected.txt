0
FFFFFFFF
FFFFFFFE
FFFFFFFD
